### PR TITLE
remove allow-untyped-defs for torch/jit/_ir_utils.py

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #143371
* #143370
* #143369
* #143368
* #143367
* __->__ #143366

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

File: torch/jit/_ir_utils.py [hunk 1]
```diff
@@ -9,18 +8,20 @@ def __init__(
...
         self.g = insert_point_graph
         self.guard = None
 
-    def __enter__(self):
+    def __enter__(self) -> None:
         self.prev_insert_point = self.g.insertPoint()
         self.g.setInsertPoint(self.insert_point)
 
-    def __exit__(self, *args):
+    def __exit__(self, *args: object) -> None:
```
Comment (aorenste) [inline torch/jit/_ir_utils.py:20]: This is fine - although technically the arg types are:
```
def __exit__(
        self,
        exc_type: type[BaseException] | None,
        exc_val: BaseException | None,
        exc_tb: TracebackType | None,
    ) -> None:
```

File: torch/jit/_ir_utils.py [hunk 2]
```diff
@@ -9,18 +8,20 @@ def __init__(
         self,
         insert_point_graph: torch._C.Graph,
         insert_point: Union[torch._C.Node, torch._C.Block],
-    ):
+    ) -> None:
```
Comment (aorenste) [inline torch/jit/_ir_utils.py:11]: nit: Technically this one shouldn't be necessary (you only need `-> None` on __init__ if it has no parameters other than `self`).

**Review by aorenste — APPROVED** [review]

File: torch/jit/_ir_utils.py (hunk 1 above)
Comment (bobrenjc93) [reply torch/jit/_ir_utils.py:20]: This and the other comment are fixed in https://github.com/pytorch/pytorch/pull/143371

**Pull request closed without merging**